apps/prairielearn/src/pages/instructorInstanceAdminSettings/instructorInstanceAdminSettings.js

```diff
@@ -17,136 +15,124 @@ import {
+
+        if (courseInstanceID === null) {
+          throw new Error('Failed to find the new course instance ID');
         }
-      });
-    });
-  } else if (req.body.__action === 'change_id') {
-    debug(`Change short_name from ${res.locals.course_instance.short_name} to ${req.body.id}`);
-    if (!req.body.id) return next(new Error(`Invalid CIID (was falsy): ${req.body.id}`));
-    if (!/^[-A-Za-z0-9_/]+$/.test(req.body.id)) {
-      return next(
-        new Error(
-          `Invalid CIID (was not only letters, numbers, dashes, slashes, and underscores, with no spaces): ${req.body.id}`,
-        ),
-      );
-    }
-    let ciid_new;
-    try {
-      ciid_new = path.normalize(req.body.id);
-    } catch (err) {
-      return next(new Error(`Invalid CIID (could not be normalized): ${req.body.id}`));
-    }
-    if (res.locals.course_instance.short_name === ciid_new) {
-      debug('The new ciid is the same as the old ciid - do nothing');
-      res.redirect(req.originalUrl);
-    } else {
-      const editor = new CourseInstanceRenameEditor({
+
+        flash(
+          'success',
+          'Course instance copied successfully. You are new viewing your copy of the course instance.',
+        );
+        res.redirect(
+          res.locals.plainUrlPrefix +
+            '/course_instance/' +
+            courseInstanceID +
+            '/instructor/instance_admin/settings',
+        );
+      } catch (err) {
+        res.redirect(res.locals.urlPrefix + '/edit_error/' + serverJob.jobSequenceId);
+      }
+    } else if (req.body.__action === 'delete_course_instance') {
+      const editor = new CourseInstanceDeleteEditor({
         locals: res.locals,
-        ciid_new: ciid_new,
       });
-      editor.canEdit((err) => {
-        if (ERR(err, next)) return;
-        editor.doEdit((err, job_sequence_id) => {
-          if (ERR(err, (e) => logger.error('Error in doEdit()', e))) {
-            res.redirect(res.locals.urlPrefix + '/edit_error/' + job_sequence_id);
-          } else {
-            res.redirect(req.originalUrl);
-          }
+
+      const serverJob = await editor.prepareServerJob();
+      try {
+        await editor.executeWithServerJob(serverJob);
+        res.redirect(`${res.locals.plainUrlPrefix}/course/${res.locals.course.id}/course_admin`);
+      } catch (err) {
+        res.redirect(res.locals.urlPrefix + '/edit_error/' + serverJob.jobSequenceId);
+      }
+    } else if (req.body.__action === 'change_id') {
+      if (!req.body.id) throw new Error(`Invalid CIID (was falsy): ${req.body.id}`);
+      if (!/^[-A-Za-z0-9_/]+$/.test(req.body.id)) {
+        throw new Error(
+          `Invalid CIID (was not only letters, numbers, dashes, slashes, and underscores, with no spaces): ${req.body.id}`,
+        );
+      }
+      let ciid_new;
+      try {
+        ciid_new = path.normalize(req.body.id);
+      } catch (err) {
+        throw new Error(`Invalid CIID (could not be normalized): ${req.body.id}`);
+      }
```

Comment:
Can we make these 400s instead of 500s?

Reply:
Updated these to 400s in [7733056](https://github.com/PrairieLearn/PrairieLearn/pull/9591/commits/77330561c88ad62551104cdfe920201c8dc11abe).